src/c_types.rs

```diff
@@ -1,18 +1,23 @@
 #![allow(non_camel_case_types)]
 
-use core::ffi;
+#[cfg(target_arch = "x86_64")]
+mod c {
+    use core::ffi;
 
-pub type c_int = i32;
-pub type c_char = i8;
-pub type c_long = i64;
-pub type c_longlong = i64;
-pub type c_short = i16;
-pub type c_uchar = u8;
-pub type c_uint = u32;
-pub type c_ulong = u64;
-pub type c_ulonglong = u64;
-pub type c_ushort = u16;
-pub type c_schar = i8;
-pub type c_size_t = usize;
-pub type c_ssize_t = isize;
-pub type c_void = ffi::c_void;
+    pub type c_int = i32;
+    pub type c_char = i8;
+    pub type c_long = i64;
+    pub type c_longlong = i64;
+    pub type c_short = i16;
+    pub type c_uchar = u8;
+    pub type c_uint = u32;
+    pub type c_ulong = u64;
+    pub type c_ulonglong = u64;
+    pub type c_ushort = u16;
+    pub type c_schar = i8;
+    pub type c_size_t = usize;
+    pub type c_ssize_t = isize;
+    pub type c_void = ffi::c_void;
```

Comment:
This can probably be generic but I have no strong opinions.

A little more generally, it'd be neat to have tests? I wonder how best to have tests. Maybe a .c file that exports sizeof and alignof each type and a .rs file that checks them, linked into the same kernel module?

Reply:
These are basically implicitly tested by every binding that uses one of these tests. More precise tests would be good, but even just having a kernel module and printing from it already uses lots of these.